src/Api/SecretsManager/Controllers/AccessPoliciesController.cs

```diff
@@ -48,21 +76,83 @@ public async Task<ProjectAccessPoliciesResponseModel> GetProjectAccessPoliciesAs
     public async Task<BaseAccessPolicyResponseModel> UpdateAccessPolicyAsync([FromRoute] Guid id,
         [FromBody] AccessPolicyUpdateRequest request)
     {
-        var result = await _updateAccessPolicyCommand.UpdateAsync(id, request.Read, request.Write);
+        var userId = _userService.GetProperUserId(User).Value;
+        var result = await _updateAccessPolicyCommand.UpdateAsync(id, request.Read, request.Write, userId);
 
         return result switch
         {
             UserProjectAccessPolicy accessPolicy => new UserProjectAccessPolicyResponseModel(accessPolicy),
             GroupProjectAccessPolicy accessPolicy => new GroupProjectAccessPolicyResponseModel(accessPolicy),
             ServiceAccountProjectAccessPolicy accessPolicy => new ServiceAccountProjectAccessPolicyResponseModel(
                 accessPolicy),
-            _ => throw new ArgumentException("Unsupported access policy type provided.")
+            _ => throw new ArgumentException("Unsupported access policy type provided."),
         };
     }
 
     [HttpDelete("{id}")]
     public async Task DeleteAccessPolicyAsync([FromRoute] Guid id)
     {
-        await _deleteAccessPolicyCommand.DeleteAsync(id);
+        var userId = _userService.GetProperUserId(User).Value;
+        await _deleteAccessPolicyCommand.DeleteAsync(id, userId);
+    }
+
+    [HttpGet("/organizations/{id}/access-policies/potential-grantees")]
+    public async Task<ListResponseModel<PotentialGranteeResponseModel>> GetPotentialGranteesAsync([FromRoute] Guid id, [FromQuery] bool includeServiceAccounts = false)
```

Comment:
This seems a bit odd, do we want two different endpoints for this? Or is there a scenario where you want both user/groups and service accounts?

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/c976495d958eef84b4cd77ce057ba69e203760f6